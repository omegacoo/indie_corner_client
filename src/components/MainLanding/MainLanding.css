.MainLanding {
    border: 1px solid black;
    height: calc(90vh - 4px);   /* Remove calc when done with design */
}

.MainLanding-title {
    border: 1px solid black;
    width: 50vw;
    margin-left: auto;
    margin-right: auto;
    margin-top: 20vh;
    text-align: center;
}

.MainLanding button {
    display: block;
    margin: 2vh;
    margin-left: auto;
    margin-right: auto;
}