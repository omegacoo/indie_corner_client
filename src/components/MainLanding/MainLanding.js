import React from 'react';
import { Link } from 'react-router-dom';

import './MainLanding.css';

export default class MainLanding extends React.Component {

    render(){
        return(
            <div className='MainLanding'>
                <h2 className='MainLanding-title'>
                    Welcome to THE forum for the Indie Gaming Community!
                </h2>

                <Link className='MainLanding-login-button'>
                    Log in
                </Link>

                <Link className='MainLanding-register-button'>
                    Register
                </Link>

                <Link className='MainLanding-anonymous-button'>
                    Go Anonymous
                </Link>
            </div>
        );
    };
};