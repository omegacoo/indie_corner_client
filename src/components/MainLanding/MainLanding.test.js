import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MainLanding from './MainLanding';

describe('MainLanding component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(
            <BrowserRouter>
                <MainLanding />
            </BrowserRouter>, 
            div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI correctly', () => {
        expect(
        renderer.create(
            <BrowserRouter>
                <MainLanding />
            </BrowserRouter>).toJSON()
        ).toMatchSnapshot();
    });
});