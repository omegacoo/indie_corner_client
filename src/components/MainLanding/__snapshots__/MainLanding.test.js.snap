// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLanding component renders UI correctly 1`] = `
<div
  className="MainLanding"
>
  <h2
    className="MainLanding-title"
  >
    Welcome to THE forum for the Indie Gaming Community!
  </h2>
  <a
    className="MainLanding-login-button"
    href=""
    onClick={[Function]}
  >
    Log in
  </a>
  <a
    className="MainLanding-register-button"
    href=""
    onClick={[Function]}
  >
    Register
  </a>
  <a
    className="MainLanding-anonymous-button"
    href=""
    onClick={[Function]}
  >
    Go Anonymous
  </a>
</div>
`;
